### PR TITLE
Move kerberos attributes to bcpc-hadoop/attributes/kerberos.rb

### PR DESCRIPTION
**Pull request by pu239ppy**
Defaults to false

### CODE REVIEW

File: cookbooks/bcpc-hadoop/attributes/default.rb [hunk 1]
```diff
@@ -126,3 +126,6 @@
 # Some jmxtrans defaults
 #
 default['jmxtrans']['run_interval'] = "15"
+
+# explicitly set kerberos enable
+node[:bcpc][:hadoop][:kerberos][:enable] = false
```
Comment (kiiranh) [inline cookbooks/bcpc-hadoop/attributes/default.rb]: why not `*default*[:bcpc][:hadoop][:kerberos][:enable] = false`

**Comment by kiiranh** [comment]
Fixes Issue #258

**Comment by pu239ppy** [comment]
Already fixed in the environment

**Comment by pu239ppy** [comment]
Moved all kerberos related attributes into kerberos.rb

File: environments/Test-Laptop.json [hunk 2]
```diff
@@ -42,31 +42,10 @@
...
-            "httpfs" : {"principal" : "httpfs", "keytab" : "httpfs.service.keytab", "owner" : "httpfs", "princhost": "_HOST", "perms": "0600", "spnego_keytab" : "httpfs.service.keytab"},
-            "hive" : {"principal" : "hive", "keytab" : "hive.service.keytab", "owner" : "hive", "princhost": "_HOST", "perms": "0600", "spnego_keytab" : "hive.service.keytab"},
-            "oozie" : {"principal" : "oozie", "keytab" : "oozie.service.keytab", "owner" : "oozie", "princhost": "_HOST", "perms": "0600", "spnego_keytab" : "oozie.service.keytab"}
-          },
-          "keytab": { 
-            "dir": "/etc/security/keytabs",
-            "recreate": false
-          }
-        },
+
```
Comment (kiiranh) [inline environments/Test-Laptop.json]: not to be pedantic, an unwanted blank line here?

File: environments/Test-Laptop.json [hunk 3]
```diff
@@ -42,31 +42,10 @@
...
         "disks" : ["sdb" , "sdc" ],
         "distribution" : {
-          "version" : "HDP",
+  "version" : "HDP",
```
Comment (kiiranh) [inline environments/Test-Laptop.json]: not to be pedantic, indentation messed up here?

File: cookbooks/bcpc-hadoop/attributes/kerberos.rb [hunk 4]
```diff
@@ -0,0 +1,17 @@
+# Kerberos settings 
+default[:bcpc][:hadoop][:kerberos][:enable] = false
+default[:bcpc][:hadoop][:kerberos][:realm] = BCPC.EXAMPLE.COM
```
Comment (kiiranh) [inline cookbooks/bcpc-hadoop/attributes/kerberos.rb]: value needs to be quotes i.e "BCPC.EXAMPLE.COM"

File: cookbooks/bcpc-hadoop/attributes/kerberos.rb [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+default[:bcpc][:hadoop][:kerberos][:hbase] = {"principal" => "hbase", "keytab" => "hbase.service.keytab", "owner" => "hbase", "princhost" => "_HOST", "perms" => "0600", "spnego_keytab" =>  "hbase.service.keytab"}
+default[:bcpc][:hadoop][:kerberos][:httpfs] = {"principal" => "httpfs", "keytab" => "httpfs.service.keytab", "owner" => "httpfs", "princhost"  => "_HOST", "perms" => "0600", "spnego_keytab" => "httpfs.service.keytab"}
+default[:bcpc][:hadoop][:kerberos][:hive] = {"principal" => "hive", "keytab" => "hive.service.keytab", "owner" => "hive", "princhost" => "_HOST", "perms" => "0600", "spnego_keytab" => "hive.service.keytab"}
+default[:bcpc][:hadoop][:kerberos][:oozie] = {"principal" => "oozie", "keytab" => "oozie.service.keytab", "owner" => "oozie", "princhost" => "_HOST", "perms" => "0600", "spnego_keytab" => "oozie.service.keytab"}
+default[:bcpc][:hadoop][:kerberos][:keytab] =  {"dir" => "/etc/security/keytabs"}
+default[:bcpc][:hadoop][:kerberos][:keytab] = {"recreate" => false}
```
Comment (kiiranh) [inline cookbooks/bcpc-hadoop/attributes/kerberos.rb]: This line will redefine the hash `default[:bcpc][:hadoop][:kerberos][:keytab]` and you will loose `{"dir" => "/etc/security/keytabs"}`. Instead define it as:
````
default[:bcpc][:hadoop][:kerberos][:keytab] = {
  "dir" => "/etc/security/keytabs",
  "recreate" => false
}
````

File: cookbooks/bcpc-hadoop/attributes/kerberos.rb [hunk 6]
```diff
@@ -0,0 +1,17 @@
+# Kerberos settings 
+default[:bcpc][:hadoop][:kerberos][:enable] = false
+default[:bcpc][:hadoop][:kerberos][:realm] = BCPC.EXAMPLE.COM
+default[:bcpc][:hadoop][:kerberos][:namenode] = {"principal" => "nn", "keytab" => "nn.service.keytab", "owner" => "hdfs", "princhost" => "_HOST", "perms"=> "0600", "spnego_keytab" => "nn.service.keytab"}
+default[:bcpc][:hadoop][:kerberos][:datanode] = {"principal" => "dn", "keytab" => "dn.service.keytab", "owner" => "hdfs", "princhost" => "_HOST", "perms"=> "0600", "spnego_keytab" => "dn.service.keytab"}
+default[:bcpc][:hadoop][:kerberos][:journalnode] =  {"principal" => "jn", "keytab" => "jn.service.keytab", "owner" => "hdfs", "princhost" => "_HOST", "perms"=> "0600", "spnego_keytab" => "jn.service.keytab"}
+default[:bcpc][:hadoop][:kerberos][:resourcemanager]  =  {"principal" => "rm", "keytab" => "rm.service.keytab", "owner" => "yarn", "princhost" => "_HOST", "perms" => "0600", "spnego_keytab" => "rm.service.keytab"}
```
Comment (kiiranh) [inline cookbooks/bcpc-hadoop/attributes/kerberos.rb]: extra spaces before and after the `=` sign

File: cookbooks/bcpc-hadoop/attributes/kerberos.rb [hunk 7]
```diff
@@ -0,0 +1,17 @@
+# Kerberos settings 
+default[:bcpc][:hadoop][:kerberos][:enable] = false
+default[:bcpc][:hadoop][:kerberos][:realm] = "BCPC.EXAMPLE.COM"
+default[:bcpc][:hadoop][:kerberos][:namenode] = {"principal" => "nn", "keytab" => "nn.service.keytab", "owner" => "hdfs", "princhost" => "_HOST", "perms"=> "0600", "spnego_keytab" => "nn.service.keytab"}
+default[:bcpc][:hadoop][:kerberos][:datanode] = {"principal" => "dn", "keytab" => "dn.service.keytab", "owner" => "hdfs", "princhost" => "_HOST", "perms"=> "0600", "spnego_keytab" => "dn.service.keytab"}
+default[:bcpc][:hadoop][:kerberos][:journalnode] =  {"principal" => "jn", "keytab" => "jn.service.keytab", "owner" => "hdfs", "princhost" => "_HOST", "perms"=> "0600", "spnego_keytab" => "jn.service.keytab"}
+default[:bcpc][:hadoop][:kerberos][:resourcemanager]  =  {"principal" => "rm", "keytab" => "rm.service.keytab", "owner" => "yarn", "princhost" => "_HOST", "perms" => "0600", "spnego_keytab" => "rm.service.keytab"}
```
Comment (kiiranh) [inline cookbooks/bcpc-hadoop/attributes/kerberos.rb]: extra space before and after `=`

File: environments/Test-Laptop.json [hunk 8]
```diff
@@ -42,32 +42,10 @@
...
-            "dir": "/etc/security/keytabs",
-            "recreate": false
-          }
-        },
         "disks" : ["sdb" , "sdc" ],
-        "distribution" : {
+          "distribution" : {
           "version" : "HDP",
-          "key" : "hortonworks.key"
+        "key" : "hortonworks.key"
```
Comment (cbaenziger) [inline environments/Test-Laptop.json]: It still looks like `distribution` is indented too far and `key` is out-dented... Perhaps comparing against a JSON pretty printer can help (e.g. `python -m json.tool environment/Test-Laptop.json`)?

**Comment by pu239ppy** [comment]
Done

**Comment by kiiranh** [comment]
Formatting looks fine now. LGTM

**Pull request merged**